### ISSUE TITLE
xcode version check is broken

### ISSUE THREAD

**Issue by kazzmir** [ios-control/ios-deploy#378]
The version check in check_reqs.js uses string comparison rather than something like semver, which causes newer versions of xcode to be considered out of date.

With xcode 10.1 or 10.2 the following check fails:
https://github.com/ios-control/ios-deploy/blob/3a1c5622dbda795c929541c913b503ed485bd975/src/scripts/check_reqs.js#L31
```
'10.1' < '7.0'
```

```
$ node
> let a = '10.1'
undefined
> let b = '7.0'
undefined
> a < b
true
```

The version check should compare the version properly by comparing the major version first, then the minor version if necessary.

**Comment by gabebear** [comment]
XCODEBUILD_MIN_VERSION is a number, not a string.

Are you hitting an error here? (I'm not arguing that the logic is wrong, just wondering if this is actively causing an issue)

```
'10.1' > 7.0
```
```
$ node
> let a = '10.1'
undefined
> let b = 7.0
undefined
> a < b
false
```

**Comment by kazzmir** [comment]
Oops sorry you are right. I was looking at the source of ios-deploy 1.9.2 where XCODEBUILD_MIN_VERSION is defined as a string
```
var XCODEBUILD_MIN_VERSION = '7.0';
```

**Comment by gabebear** [comment]
I think the current bad cases are when XCode has a 3-part major.minor.patch version:
```
> '10.3.2' > 7.0
false
```
I think we should just remove this check(XCode 7.0 is very old) #382

**Comment by shazron** [comment]
Fixed in #382
